app/bundles/LeadBundle/Entity/StatDeviceRepository.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * @package     Mautic
+ * @copyright   2014 Mautic Contributors. All rights reserved.
+ * @author      Mautic
+ * @link        http://mautic.org
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Entity;
+
+use Doctrine\ORM\Tools\Pagination\Paginator;
+use Mautic\CoreBundle\Entity\CommonRepository;
+use Mautic\CoreBundle\Helper\DateTimeHelper;
+
+/**
+ * Class StatRepository
+ *
+ * @package Mautic\EmailBundle\Entity
+ */
+class StatDeviceRepository extends CommonRepository
+{
+    /**
+     * @param           $emailIds
+     * @param \DateTime $fromDate
+     *
+     * @return array
+     */
+    public function getDeviceStats($statIds, $listId= null,  \DateTime $fromDate = null, \DateTime $toDate = null, $channel = null, $channelId = null)
+    {
+        $inIds = (!is_array($statIds)) ? array($statIds) : $statIds;
+
+        if(empty($inIds))
+        {
+            return array();
+        }
+        $sq = $this->_em->getConnection()->createQueryBuilder();
+        $sq->select('count(es.id) as count, es.device as device')
+            ->from(MAUTIC_TABLE_PREFIX.'lead_stats_devices', 'es');
+        if ($statIds) {
+            if (!is_array($statIds)) {
+                $statIds = array((int) $statIds);
+            }
+            $sq->where(
+                $sq->expr()->in('es.stat_id', $statIds)
+            );
+        }
+
+        if ($channel) {
+            $sq->where(
+                $sq->expr()->eq('es.channel', $channel)
```

Comment:
$channel needs to be set in `setParameter()` or else $channel will be used as a column name rather than value.